### PR TITLE
fix: Update sceenshoting mechanism to work with Xcode 13.beta

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Apple did deprecate the way we were previously taking screenshots. Lets use the new API now to make it working

### CODE REVIEW

**Comment by mykola-mokhnach** [comment]
@KazuCocoa @Dan-Maor  Could you please check if the algorithm properly works for a real device running iOS below v15, while major Xcode version is 13?

**Comment by KazuCocoa** [comment]
Seems good. I tested the below combinations on real devices.

Xcode 10.3 x iOS 12.4
Xcode 12.4 x iOS 12.4,  iOS 14.5
Xcode 13-beta x iOS 12.4, iOS 14.5

Will take a look at code a bit later

File: PrivateHeaders/XCTest/XCTestManager_ManagerInterface-Protocol.h [hunk 1]
```diff
@@ -47,4 +47,7 @@
 - (void)_XCT_requestScreenshotOfScreenWithID:(unsigned int)arg1 withRect:(struct CGRect)arg2 uti:(NSString *)arg3 compressionQuality:(double)arg4 withReply:(void (^)(NSData *, NSError *))arg5;
 - (void)_XCT_requestScreenshotOfScreenWithID:(unsigned int)arg1 withRect:(struct CGRect)arg2 withReply:(void (^)(NSData *, NSError *))arg3;
 - (void)_XCT_requestSnapshotForElement:(XCAccessibilityElement *)arg1 attributes:(NSArray *)arg2 parameters:(NSDictionary *)arg3 reply:(void (^)(XCElementSnapshot *, NSError *))arg4;
+
+// Available since Xcode 12.5. Required (!!!) to use since Xcode 13
+- (void)_XCT_requestScreenshot:(/*XCTScreenshotRequest * */id)arg1 withReply:(void (^)(/*XCTImage * */id, NSError *))arg2;
```
Comment (KazuCocoa) [inline PrivateHeaders/XCTest/XCTestManager_ManagerInterface-Protocol.h:52]: oh, 12.5.
I just compared the diff between 13 and 12.5... (so I found no API diff about this point..)

**Review by KazuCocoa — APPROVED** [review]

**Comment by KazuCocoa** [comment]
Some test failed in xcuitest driver test cases -> https://github.com/appium/appium-xcuitest-driver/issues/1312
ruby_lib_core suite was a background process in tvOS, but it seemed simulator or tvOS side thing.

So, as WDA, this change seems enough

**Review by Dan-Maor — APPROVED** [review]

**Comment by jlipps** [comment]
wow obj-c is painful to read

**Pull request merged**